### PR TITLE
Add support for OSD elements variants

### PR DESCRIPTION
**Pull request by McGiverGim**
Fixes https://github.com/betaflight/betaflight-configurator/issues/2398

Adds support for different variants at the OSD elements.

It includes three elements with variants:
- Battery usage. Added in firmware in original PR: https://github.com/betaflight/betaflight/pull/10517
- Altitude: Added in firmware in PR: https://github.com/betaflight/betaflight/pull/10551
- GPS latitude and longitude: Added in firmware in PR: https://github.com/betaflight/betaflight/pull/10566 ~~(WARNING: This PR has not been merged at this moment).~~

image

NOTE: I've not tested it with a real VTX (I don't have one at this moment with I can use to make tests), only tested using the Configurator preview, so it will be great if someone can make some test...

### CODE REVIEW

**Comment by McGiverGim** [comment]
If someone wants to test it, at this moment can be downloaded from Azure here: https://dev.azure.com/Betaflight/Betaflight%20Nightlies/_build/results?buildId=2405&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by haslinghuis** [comment]
@McGiverGim tested `Battery Usage`.  (at the moment have no `baro` or `gps` at hand).

It does change the OSD element on my FPV monitor and represent the selected variants, but the reading/output is `zero`  for all variants, while `battery average cell voltage`, `battery current draw` and `battery current mAh` draw do show the expected values.

**Comment by McGiverGim** [comment]
Strange, if it changes the variant on the googles, it must be working. Maybe a bug in the firmware part? Could you post the diff values of the osd element? To see if I'm sending the correct value or not.

**Comment by haslinghuis** [comment]
```
set osd_mah_drawn_pos = 2282
set osd_altitude_pos = 18502
set osd_avg_cell_voltage_pos = 2218
set osd_battery_usage_pos = 34950
```

**Comment by haslinghuis** [comment]
image

**Comment by McGiverGim** [comment]
One silly question... Have you defined the capacity of yours batteries, true?

**Comment by McGiverGim** [comment]
@haslinghuis thanks for the testing!

**Comment by McGiverGim** [comment]
@mikeller if you have some time, I would like this to be merged. The firmware part was merged some days ago so this is ready to go.

**Review by haslinghuis — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - Github identifies PR as mergeable -> PASS
 - PR is not in draft -> PASS
 - Assigned to a milestone -> FAIL
 - cooling off period lapsed (3/06/2021 21:21:33) -> PASS
 - commit count less or equal to three (1 commits) -> PASS
 - 'Don't merge' label NOT found -> PASS
 - 'Needs coordination with betaflight' label NOT found -> FAIL
 - At least one 'RN:' label found -> FAIL
 - 'Tested' label found -> FAIL
 - assigned to an approver (None) -> FAIL
 - approver count at least three (1 approvers) -> FAIL

**Review by Asizon — APPROVED** [review]

File: src/js/tabs/osd.js
Comment (chmelevskij) [inline src/js/tabs/osd.js:325]: I think pulling out those symbols to vars even in the same scope would make it a bit more legible.
```suggestion
    let value;
    const start = FONT.symbol(SYM.PB_START);
    const full = FONT.symbol(SYM.PB_FULL);
    const end = FONT.symbol(SYM.PB_END);
    const empty =  FONT.symbol(SYM.PB_EMPTY);
    const close = FONT.symbol(SYM.PB_CLOSE);
    switch (variantSelected) {
        case 0:
            value = start + full + full + full + full + full + full + full + full + full + end + empty + close;
            break;

        case 1:
            value = start + full + full + full + full + full + end + empty + empty + empty + empty + empty + close;
            break;
```

**Review by chmelevskij — APPROVED** [review]
lgtm

File: src/js/tabs/osd.js
Comment (haslinghuis) [reply src/js/tabs/osd.js:325]: Can beautify further with .repeat(count)

Comment (McGiverGim) [reply src/js/tabs/osd.js:325]: I like the idea... But the weekend I'm out... I will do a new PR later

**Pull request merged**
